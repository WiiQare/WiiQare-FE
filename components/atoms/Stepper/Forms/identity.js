import React, { useContext, useState } from "react";
import { ErrorMessage, Field, Form, Formik } from "formik";
import { FormContext } from "../../../../pages/voucher/buy";
import * as yup from "yup";
import { HiArrowSmRight, HiOutlineInformationCircle } from "react-icons/hi";
import { CountryDropdown } from 'react-country-region-selector';

import 'react-phone-number-input/style.css'
import PhoneInput from 'react-phone-number-input'

function Identity() {
	const { activeStepIndex, setActiveStepIndex, formData, setFormData } = useContext(FormContext);
	const [phone, setPhone] = useState()

	const renderError = (message) => (
		<p className="text-xs text-red-600 font-light flex items-center gap-1"><HiOutlineInformationCircle />{message}</p>
	);

	const ValidationSchema = yup.object().shape({
		name: yup.string().required("Fistname is a required field"),
		lastname: yup.string().required("Lastname is a required field"),
		email: yup.string().email().required("Address email is a required field"),
		address: yup.string().required(),
		address2: yup.string(),
		phone: yup.string().required("Phone"),
	});

	console.log(phone);

	return (
		<Formik
			initialValues={{
				name: "",
				lastname: "",
				email: "",
				address: "",
				address2: "",
				phone: ""

			}}
			validationSchema={ValidationSchema}
			onSubmit={(values) => {

				values.phone = phone;
				const data = { ...formData, ...values };
				setFormData(data);
				setActiveStepIndex(activeStepIndex + 1);
			}}
		>
			<Form className="flex flex-col justify-start md:w-2/3 gap-4">
				<div className="text-2xl font-medium my-4 capitalize">Enter Identity for receiver !</div>

				<div className="space-y-1">
					<div className="flex flex-col items-start space-y-1">
						<label className="text-gray-500 font-light text-sm">Phone Number</label>
						<PhoneInput
							defaultCountry="CD"
							placeholder="Enter phone number"
							value={phone}
							onChange={setPhone}
							name="phone"
							className="rounded-lg border px-4 py-1.5 w-full placeholder:text-gray-400 text-gray-700 text-md"	
						/>
					</div>
					<ErrorMessage name="phone" render={renderError} />

				</div>

				<div className="grid md:grid-cols-2 gap-4">
					<div className="space-y-1">
						<div className="flex flex-col items-start gap-1">
							<label className="text-gray-500 font-light text-sm">Firstname</label>
							<Field
								name="name"
								className="rounded-lg border px-4 py-3 w-full placeholder:text-gray-400 text-gray-700 text-md"
								placeholder="John"
							/>
						</div>
						<ErrorMessage name="name" render={renderError} />
					</div>

					<div className="space-y-1">
						<div className="flex flex-col items-start gap-1">
							<label className="text-gray-500 font-light text-sm">Lastname</label>
							<Field
								name="lastname"
								className="rounded-lg border px-4 py-3 w-full placeholder:text-gray-400 text-gray-700 text-md"
								placeholder="Doe"
							/>
						</div>
						<ErrorMessage name="lastname" render={renderError} />
					</div>

				</div>

				<div className="space-y-1">
					<div className="flex flex-col items-start space-y-1">
						<label className="text-gray-500 font-light text-sm">Email</label>
						<Field
							name="email"
							className="rounded-lg border px-4 py-3 w-full placeholder:text-gray-400 text-gray-700 text-md"
							placeholder="john.doe@wiiqare.com"
						/>
					</div>
					<ErrorMessage name="email" render={renderError} />
				</div>


				<div className="grid md:grid-cols-2 gap-4">
					<div className="space-y-1">
						<div className="flex flex-col items-start gap-1">
							<label className="text-gray-500 font-light text-sm">Address 1</label>
							<Field
								name="address"
								className="rounded-lg border px-4 py-3 w-full placeholder:text-gray-400 text-gray-700 text-md"
								placeholder="23, Mimosas C/Ngaliema"
							/>
						</div>
						<ErrorMessage name="address" render={renderError} />
					</div>

					<div className="space-y-1">
						<div className="flex flex-col items-start gap-1">
							<label className="text-gray-500 font-light text-sm">Address 2 <span className="text-gray-400">(optional)</span></label>
							<Field
								name="address2"
								className="rounded-lg border px-4 py-3 w-full placeholder:text-gray-400 text-gray-700 text-md"
								placeholder="23, Mimosas C/Ngaliema"
							/>
						</div>
						<ErrorMessage name="address2" render={renderError} />
					</div>

				</div>

				<div className="flex flex-row-reverse ">

					<button
						className="bg-primary flex gap-3 items-center w-fit font-medium text-white my-2 py-3 px-5 hover:bg-blue-500 duration-200 transition-all hover:shadow rounded-lg"
						type="submit"
					>
						Next <HiArrowSmRight />
					</button>

				</div>
			</Form>
		</Formik>
	);
}

export default Identity;
