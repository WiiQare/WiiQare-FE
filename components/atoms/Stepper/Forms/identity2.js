import React, { createContext, useContext, useEffect, useState } from "react";
import { ErrorMessage, Field, FormikProvider, useFormik } from "formik";
import { FormContext } from "../../../../pages/voucher/buy";
import * as yup from "yup";
import {
  HiArrowSmLeft,
  HiArrowSmRight,
  HiOutlineInformationCircle,
} from "react-icons/hi";
import { useSession } from "next-auth/react";
import { FiChevronRight, FiUserPlus } from "react-icons/fi";

import "react-phone-number-input/style.css";
import PhoneInput from "react-phone-number-input";
import Fetcher from "../../../../lib/Fetcher";
import { autocomplete, savePatient } from "../../../../lib/helper";
import { useMutation } from "react-query";
import LoadingButton from "../../Loader/LoadingButton";
import { useDispatch, useSelector } from "react-redux";
import { setPatientDispatch } from "../../../../redux/reducer";
import Toast from "../../Toast";
import MuiPhoneNumber from "material-ui-phone-number";
import { Stack, TextField } from "@mui/material";
import { Splide, SplideSlide, SplideTrack } from "@splidejs/react-splide";
import "@splidejs/react-splide/css";
import avatar from "../../../../public/images/femme.png";
import Image from "next/image";
import { CiCircleCheck, CiCircleInfo } from "react-icons/ci";
import { BiCaretRight } from "react-icons/bi";
import CountrySelect from "../../Input/Country";
export const CountryContext = createContext();

function Identity2() {
  const { activeStepIndex, setActiveStepIndex, formData, setFormData } =
    useContext(FormContext);
  const [phone, setPhone] = useState();
  const [patient, setPatient] = useState({});
  const [patientExist, setPatientExist] = useState(false);
  const [requestComplete, setRequestComplete] = useState(false);
  const { data: session } = useSession();
  const dispatch = useDispatch();
  const [state, setState] = useState({ type: 0, message: "" });
  const client = useSelector((state) => state.app.client);
  const [newBenecifiare, setNewBenecifiare] = useState(false);
  const [activeIndexSlide, setActiveIndexSlide] = useState(null);
  const [country, setCountry] = useState("cd");
  const [countryLabel, setCountryLabel] = useState("RD Congo");
  const [allBeneficiare, setAllBeneficiare] = useState([]);
  const [tempBeneficiare, setTempBeneficiare] = useState([]);
  const [dial, setDial] = useState("243");

  const renderError = (message) => (
    <p className="text-xs text-red-600 font-light flex items-center gap-1">
      <HiOutlineInformationCircle />
      {message}
    </p>
  );

  const ValidationSchema = yup.object().shape({
    firstName: yup.string().required("Le nom est un champ obligatoire"),
    lastName: yup.string().required("Le prénom est un champ obligatoire"),
    email: yup.string().email(),
    homeAddress: yup
      .string()
      .required("L'adresse du domicile est un champ obligatoire"),
    city: yup.string().required("La ville est un champ obligatoire"),
    phoneNumber: yup.string().required("Téléphone est requis"),
    country: yup.string(),
  });

  const savePatientMutation = useMutation(savePatient, {
    onSuccess: (res) => {
      if (res.code) {
        setState({ type: 2, message: res.message ?? res.description });
        setTimeout(() => {
          setState({ type: 0, message: "" });
        }, 3000);
      } else {
        setState({ type: 1, message: "Enregistré avec succès" });
        dispatch(setPatientDispatch({ ...res, countryLabel }));

        setActiveStepIndex(activeStepIndex + 1);
      }
    },
  });

  const onSubmit = (values) => {
    if (Object.keys(values).length == 0) return console.log("Pas de données");

    !values.email ? delete values.email : null;

    const data = { ...formData, ...values };

    if (patientExist) {
      dispatch(
        setPatientDispatch({ ...client.patient, ...data, countryLabel })
      );
      setActiveStepIndex(activeStepIndex + 1);
    } else {
      savePatientMutation.mutate({ ...data, accessToken: session.accessToken });
    }
  };

  const closeToast = () => {
    setState({ type: 0, message: "" });
  };

  const formik = useFormik({
    initialValues: {
      firstName: localStorage.getItem("firstName") ?? "",
      lastName: localStorage.getItem("lastName") ?? "",
      email: localStorage.getItem("email") ?? "",
      homeAddress: localStorage.getItem("homeAddress") ?? "",
      city: localStorage.getItem("city") ?? "",
      phoneNumber: localStorage.getItem("phoneNumber") ?? "",
      country: localStorage.getItem("country") ?? "",
    },
    validationSchema: ValidationSchema,
    onSubmit,
  });

  const handlePhone = async (value) => {
    setPhone(value);
    setRequestComplete(true);

    const res = await autocomplete(value, session.accessToken);

    setRequestComplete(false);

    if (!res.code) {
      setPatient(res);
      setPatientExist(true);

      console.log(res);

      //Set Value
      formik.setFieldValue("phoneNumber", res.phoneNumber);
      formik.setFieldValue("firstName", res.firstName);
      formik.setFieldValue("lastName", res.lastName);
      formik.setFieldValue("email", res.email);

      dispatch(setPatientDispatch({ ...res }));
    } else {
      setPatientExist(false);
    }
  };

  const handleSelect = (item, index) => {
    console.log(item);
    setActiveIndexSlide(index);
    dispatch(
      setPatientDispatch({
        ...client.patient,
        id: item.id,
        country: item.country,
        firstName: item.firstName,
        lastName: item.lastName,
        email: item.email,
        phoneNumber: item.phoneNumber,
        city: item.city,
        homeAddress: item.homeAddress,
      })
    );
    setActiveStepIndex(activeStepIndex + 1);
  };

  const { data, isLoading, isError } = Fetcher(
    `/payer/patient?payerId=${session.user.data.userId}`,
    session.user.data.access_token
  );

  useEffect(() => {
    setAllBeneficiare(data);
    setTempBeneficiare(data);
  }, [data]);

  const handleKey = (e) => {
    let value =
      e.target.value.length > 0 && e.target.value[0] == "+"
        ? e.target.value.split("+")[1]
        : e.target.value;
    let filter = tempBeneficiare.filter(function (beneficiare) {
      return (
        new RegExp(value, "i").test(
          beneficiare.firstName + " " + beneficiare.lastName
        ) ||
        new RegExp(value, "i").test(beneficiare.phoneNumber) ||
        new RegExp(value, "i").test(beneficiare.email)
      );
    });

    setTempBeneficiare(value.trim() != "" ? filter : allBeneficiare);
  };

  return (
    <>
      {state.type > 0 ? (
        state.type == 2 ? (
          <Toast type={"danger"} message={state.message} close={closeToast} />
        ) : state.type == 1 ? (
          <Toast type={"success"} message={state.message} close={closeToast} />
        ) : (
          <></>
        )
      ) : (
        <></>
      )}

      <div className="flex flex-col justify-start md:w-2/3 w-full gap-4">
        {!newBenecifiare ? (
          <>
            {allBeneficiare && allBeneficiare.length > 0 ? (
              <>
                <div className="space-y-1 w-full">
                  <TextField
                    id="outlined-basic"
                    fullWidth
                    label="Rechercher un bénéficiaire par Nom ou par Numéro Téléphone"
                    variant="outlined"
                    onChange={handleKey}
                  />
                  {formik.errors.email && formik.touched.email ? (
                    <span className="flex items-center gap-1 text-rose-500 text-left text-xs px-1">
                      <HiOutlineInformationCircle />
                      <span>{formik.errors.email}</span>
                    </span>
                  ) : (
                    <></>
                  )}
                </div>
                <div className="flex mt-10 flex-col gap-4 w-80 md:w-full">
                  <h2 className="font-semibold text-xl">Mes bénéficiaires :</h2>

                  <div className="bg-white py-4 px-2 relative rounded-lg space-y-8 h-full w-80 md:w-full">
                    {isLoading || isError ? (
                      <div className="flex flex-col gap-2 items-center justify-center">
                        <div
                          className="animate-spin inline-block w-4 h-4 border-[2px] border-current border-t-transparent text-gray-500 rounded-full"
                          role="status"
                          ariaLabel="loading"
                        >
                          <span className="sr-only">Loading...</span>
                        </div>
                      </div>
                    ) : (
                      <>
                        {tempBeneficiare.length > 0 ? (
                          <Splide
                            hasTrack={false}
                            aria-label="Attribution"
                            options={{
                              type: "slide",
                              perPage: 2,
                              mediaQuery: "min",
                              gap: 10,
                              breakpoints: {
                                1024: {
                                  perPage: 3,
                                  gap: 20,
                                },
                              },
                              pagination: false,
                              focus: "center",
                            }}
                            onActive={(splide, slide) => {
                              if (slide.slideIndex == -1) {
                                //setActiveIndexSlide(slide.index)
                              }
                            }}
                            className="container mx-auto px-8"
                          >
                            <SplideTrack hasTrack={false}>
                              {tempBeneficiare.map((item, index) => (
                                <SplideSlide
                                  key={index}
                                  className="w-min flex flex-col gap-2 items-center justify-center p-3"
                                  onClick={() => handleSelect(item, index)}
                                >
                                  <div className="w-20 h-2O relative">
                                    <Image
                                      src={
                                        index % 2 ? avatar : "/images/homme.png"
                                      }
                                      className="object-cover rounded-xl"
                                      width={80}
                                      height={80}
                                      alt="Beneficiare image"
                                    />
                                    <span
                                      className={`${
                                        activeIndexSlide === index
                                          ? ""
                                          : "hidden"
                                      } p-1.5 rounded-lg bg-blue-600 text-white absolute right-0 bottom-0`}
                                    >
                                      <CiCircleCheck size={18} />
                                    </span>
                                  </div>

                                  <span className="font-semibold text-sm">
                                    {item.firstName} {item.lastName}
                                  </span>
                                  <span className="text-xs font-light">
                                    {item.phoneNumber}
                                  </span>
                                </SplideSlide>
                              ))}
                            </SplideTrack>

                            <div className="splide__arrows">
                              <button className="splide__arrow splide__arrow--prev bg-transparent relative !-left-7 top-2 bottom-0 !bg-[#F0F4FD] p-4 text-3xl focus:ring-0">
                                <span className="bg-white rounded-full p-1 !text-red-500">
                                  <BiCaretRight />
                                </span>
                              </button>
                              <button className="splide__arrow splide__arrow--next bg-transparent relative !-right-7 top-2 bottom-0 !bg-[#F0F4FD] p-4 text-3xl focus:ring-0">
                                <span className="bg-white rounded-full p-1 !text-red-500">
                                  <BiCaretRight />
                                </span>
                              </button>
                            </div>
                          </Splide>
                        ) : (
                          <div className="w-full flex flex-col items-center gap-3">
                            <Image
                              src="/images/box.png"
                              alt="Box image"
                              loading="lazy"
                              className="h-44 opacity-50"
                            />
                            <span className="text-gray-400 text-xs font-normal">
                              Aucun bénéficiaire en rapport avec votre
                              recherche...
                            </span>
                          </div>
                        )}
                      </>
                    )}
                  </div>
                </div>
                <div className="divider mt-8">
                  <span className="text-gray-400">OU</span>
                </div>
              </>
            ) : (
              <></>
            )}

            <div className="mt-6">
              <div
                className="w-full bg-gray-100 py-4 px-6 border-2 border-dashed border-gray-700 rounded-lg cursor-pointer hover:shadow-md hover:bg-gray-200 flex justify-between"
                onClick={() => setNewBenecifiare(true)}
              >
                <div className="flex gap-5 items-center">
                  <FiUserPlus size={26} />
                  <h4 className="font-semibold md:text-lg text-sm">
                    Ajouter un bénéficiaire
                  </h4>
                </div>
                <FiChevronRight size={26} />
              </div>
            </div>
          </>
        ) : (
          <FormikProvider value={formik}>
            <form onSubmit={formik.handleSubmit} autoComplete="false">
              <div className="flex flex-col gap-1">
                <div className="flex flex-col md:flex-row gap-6 w-full">
                  <div className="md:w-1/3 w-full inline-flex">
                    <CountryContext.Provider
                      value={{ country, setCountry, setCountryLabel }}
                    >
                      <CountrySelect />
                    </CountryContext.Provider>
                  </div>
                  <div className="flex items-center gap-1 md:w-2/3">
                    <div className="flex flex-col w-full gap-1">
                      <MuiPhoneNumber
                        fullWidth
                        label="Numéro de Téléphone"
                        variant="outlined"
                        countryCodeEditable={false}
                        select={false}
                        value={localStorage.getItem("phoneNumber") ?? ""}
                        defaultValue={localStorage.getItem("phoneNumber") ?? ""}
                        onChange={(value, country) => {
                          formik.setFieldValue("phoneNumber", value);
                          formik.setFieldValue("country", country.countryCode);
                          setDial(country.dialCode);
                          localStorage.setItem("phoneNumber", value);
                          localStorage.setItem("country", country.countryCode);
                        }}
                        defaultCountry={country}
                        name="phoneNumber"
                      />
                      {formik.values.phoneNumber.trim(" ") == "" ||
                      formik.values.phoneNumber
                        .replace("+" + dial, " ")
                        .trim(" ") == "" ? (
                        renderError("Entrez le numéro de téléphone")
                      ) : (
                        <></>
                      )}
                    </div>
                    <span className="w-fit h-fit p-2 rounded-lg flex items-center gap-2 text-sm text-gray-600">
                      <span
                        className="tooltip tooltip-bottom text-xs"
                        data-tip={`${
                          formik.values.phoneNumber.trim(" ") != ""
                            ? formik.values.phoneNumber
                            : "Ce numéro de téléphone"
                        } devra être le numéro disponible, pour être utilisé à l'hôpital `}
                      >
                        <CiCircleInfo size={23} className="text-red-400" />
                      </span>
                    </span>
                  </div>
                </div>
              </div>

              <Stack spacing={1.5} className="w-full mt-3">
                <div className="flex gap-2 md:gap-6 w-full">
                  <div className="flex flex-col gap-1 w-full">
                    <TextField
                      id="outlined-basic"
                      fullWidth
                      label="Nom de famille"
                      variant="outlined"
                      name="firstName"
                      defaultValue={localStorage.getItem("firstName") ?? ""}
                      onKeyUp={(e) => {
                        localStorage.setItem("firstName", e.target.value);
                      }}
                      autoComplete="off"
                      {...formik.getFieldProps("firstName")}
                    />

                    {formik.errors.firstName && formik.touched.firstName ? (
                      renderError(formik.errors.firstName)
                    ) : (
                      <></>
                    )}
                  </div>

                  <div className="flex flex-col gap-1 w-full">
                    <TextField
                      id="outlined-basic"
                      fullWidth
                      label="Prénom"
                      variant="outlined"
                      name="lastName"
                      defaultValue={localStorage.getItem("lastName") ?? ""}
                      onKeyUp={(e) => {
                        localStorage.setItem("lastName", e.target.value);
                      }}
                      {...formik.getFieldProps("lastName")}
                    />
                    {formik.errors.lastName && formik.touched.lastName ? (
                      renderError(formik.errors.lastName)
                    ) : (
                      <></>
                    )}
                  </div>
                </div>

                <div className="space-y-1 w-full">
                  <TextField
                    id="outlined-basic"
                    fullWidth
                    label="Adresse e-mail (optional)"
                    variant="outlined"
                    name="email"
                    defaultValue={localStorage.getItem("email") ?? ""}
                    onKeyUp={(e) => {
                      localStorage.setItem("email", e.target.value);
                    }}
                    {...formik.getFieldProps("email")}
                  />
                  {formik.errors.email && formik.touched.email ? (
                    <span className="flex items-center gap-1 text-rose-500 text-left text-xs px-1">
                      <HiOutlineInformationCircle />
                      <span>{formik.errors.email}</span>
                    </span>
                  ) : (
                    <></>
                  )}
                </div>

                <div className="flex flex-col md:flex-row gap-2 md:gap-6 w-full">
                  <div className="flex flex-col gap-1 w-full">
                    <TextField
                      id="outlined-basic"
                      fullWidth
                      label="Adresse du domicile"
                      variant="outlined"
                      name="homeAddress"
                      defaultValue={localStorage.getItem("homeAddress") ?? ""}
                      onKeyUp={(e) => {
                        localStorage.setItem("homeAddress", e.target.value);
                      }}
                      {...formik.getFieldProps("homeAddress")}
                    />

                    {formik.errors.homeAddress && formik.touched.homeAddress ? (
                      renderError(formik.errors.homeAddress)
                    ) : (
                      <></>
                    )}
                  </div>

                  <div className="flex flex-col gap-1 w-full">
                    <TextField
                      id="outlined-basic"
                      fullWidth
                      label="Ville"
                      variant="outlined"
                      name="city"
                      defaultValue={localStorage.getItem("city") ?? ""}
                      onKeyUp={(e) => {
                        localStorage.setItem("city", e.target.value);
                      }}
                      {...formik.getFieldProps("city")}
                    />
                    {formik.errors.city && formik.touched.city ? (
                      renderError(formik.errors.city)
                    ) : (
                      <></>
                    )}
                  </div>
                </div>
              </Stack>

              <div className="flex flex-row-reverse justify-between">
                <button
                  className="bg-primary flex gap-3 items-center w-fit font-medium text-white my-2 py-3 px-5 hover:bg-blue-500 duration-200 transition-all hover:shadow rounded-lg"
                  type="submit"
                >
                  {savePatientMutation.isLoading ? (
                    <LoadingButton />
                  ) : (
                    <>
                      {patientExist ? "Continuer avec ce patient" : "Suivant"}{" "}
                      <HiArrowSmRight />
                    </>
                  )}
                </button>

                <button
                  className="bg-primary flex gap-3 items-center w-fit font-medium text-white my-2 py-3 px-5 hover:bg-blue-500 duration-200 transition-all hover:shadow rounded-lg"
                  type="button"
                  onClick={() => setNewBenecifiare(false)}
                >
                  <HiArrowSmLeft /> Précédent
                </button>
              </div>
            </form>
          </FormikProvider>
        )}
      </div>
    </>
  );
}

export default Identity2;
