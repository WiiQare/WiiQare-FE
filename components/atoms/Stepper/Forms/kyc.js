import React, { useContext, useEffect, useState } from 'react';
import { FormContext } from '../../../../pages/voucher/buy';
import { useRouter } from 'next/router';
import { checkKyc, setKyc } from '../../../../lib/helper';
import { useSession } from 'next-auth/react';

function KYC() {
  const [statusID, setStatusID] = useState('IN_PROGRESS');
  const [resultKYC, setResultKYC] = useState(null);
  const {
    activeStepIndex,
    setActiveStepIndex,
    formData,
    setFormData,
    kycTest,
    setKycTest,
  } = useContext(FormContext);
  const { data: session } = useSession();
  const router = useRouter();

  const checkStatus = async () => {
    try {
      const response = fetch('/api/authologic/check', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          id: router.query.conversation,
        }),
      });

      const jsonData = (await response).json();

      jsonData
        .then((data) => {

          if (data.status != 'FINISHED') {
            setTimeout(checkStatus, 8000);
          } else {
            setStatusID("FINISHED")
            if (data.result.identity.status != "FAILED") {
              setKyc({ accessToken: session.accessToken, expire: "2023/05", cardID: "ABCD", birthday: "1990-01-01", kyc: true })
              setKycTest(false)
            } else {
              setResultKYC(data.result)
            }
          }
        })
        .catch((e) => setTimeout(checkStatus, 8000));
    } catch (error) {
      setTimeout(checkStatus, 8000);
    }
  };

  const conversation = () => {
    if (!router.query.conversation) {
      fetch('/api/authologic', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          path: router.asPath + '?step=1',
        }),
      })
        .then((res) => res.json())
        .then((data) => {
          router.push(data.url);
        })
        .catch((e) => console.log(e));
    } else {
      checkStatus();
    }
  }

  useEffect(() => {

    if ("geolocation" in navigator) {
      navigator.geolocation.getCurrentPosition(async (position) => {
        const latitude = position.coords.latitude;
        const longitude = position.coords.longitude;

        try {
          const response = await fetch(`https://api.bigdatacloud.net/data/reverse-geocode-client?latitude=${latitude}&longitude=${longitude}&localityLanguage=fr`);
          const data = await response.json();

          console.log(data.continent);

          if (/Afrique/i.test(data.continent)) {
            setKycTest(false)

          } else {
            checkKyc({ accessToken: session.accessToken }).then((data) => {
              if (data) {
                setKycTest(false)
              } else {
                conversation()
              }
            }).catch((error) => conversation())
          }
        } catch (error) {
          checkKyc({ accessToken: session.accessToken }).then((data) => {
            if (data) {
              setKycTest(false)
            } else {
              conversation()
            }
          }).catch((error) => conversation())
        }
      }, (error) => {
        checkKyc({ accessToken: session.accessToken }).then((data) => {
          if (data) {
            setKycTest(false)
          } else {
            conversation()
          }
        }).catch((error) => conversation())

      });
    } else {
      checkKyc({ accessToken: session.accessToken }).then((data) => {
        if (data) {
          setKycTest(false)
        } else {
          conversation()
        }
      }).catch((error) => conversation())
    }

  }, []);

  return (
    <div className="flex flex-col gap-6 justify-center items-center">
      {!router.query.conversation ? (
        <>{"Verification d'identité..."}</>
      ) : (
        <>
          {statusID == 'IN_PROGRESS' ? (
            <>{"Vérification d'identité en cours"}</>
          ) : (
            <>
              {statusID != 'FINISHED' ? (
                <>{'Identity vérification failed'}</>
              ) : (
                <>
                  {
                    resultKYC?.identity?.status == "FAILED" ? (
                      <div className='flex flex-col items-center justify-center gap-6 text-center'>
                        <img src='https://i.goopics.net/2askzc.png' alt='Mismatch' className='w-48 opacity-90' />
                        <span className='text-gray-400 text-sm'>{resultKYC?.identity?.errors.toString()}</span>
                      </div>
                    ) : ''
                  }
                </>
              )}
            </>
          )}
        </>
      )}
    </div>
  );
}

export default KYC;
