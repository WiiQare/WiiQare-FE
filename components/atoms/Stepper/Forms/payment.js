import React, { useContext, useState } from "react";
import { ErrorMessage, Field, Form, Formik } from "formik";
import { FormContext } from "../../../../pages/voucher/buy";
import { HiOutlineInformationCircle } from "react-icons/hi";
import * as yup from "yup";
import StripePayment from "../../../molecules/Stripe";


function Payment(props) {
	const { activeStepIndex, setActiveStepIndex, formData, setFormData } = useContext(FormContext);
	const [amount, setAmount] = useState(0);

	return (
		<>
			{/* <div className="text-2xl font-medium my-4 capitalize">Choice your payment method !</div> */}

			{amount == 0 ? <Amount amount={amount} setAmount={setAmount} /> : <StripePayment amount={amount}/>}
		</>
	);
}

export default Payment;

function Amount({ amount, setAmount }) {

	const ValidationSchema = yup.object().shape({
		amount: yup.number().required("Please enter valid amount")
	});

	const renderError = (message) => (
		<p className="text-xs text-red-600 font-light flex items-center gap-1"><HiOutlineInformationCircle />{message}</p>
	);

	return (
		<Formik
			initialValues={{
				amount: ''
			}}
			validationSchema={ValidationSchema}
			onSubmit={(values) => {
				setAmount(values.amount)
			}}
		>
			<Form className="flex justify-center md:w-2/4 w-full  py-4 items-end">
				<div className="space-y-1 md:w-3/4">
					<div className="flex items-end gap-4">
						<div className="flex flex-col items-center space-y-1">
							<Field
								className="rounded-lg border px-2 md:px-4 py-3 w-full placeholder:text-gray-400 text-gray-700 text-md"
								placeholder="Enter amount"
								min="10"
								name="amount"
								defaultValue={10}

							/>
						</div>

						<button type="submit" className="bg-primary flex gap-3 items-center w-fit font-medium text-white py-3 md:px-5 px-2 text-sm md:text-md hover:bg-blue-500 duration-200 transition-all hover:shadow rounded-lg">
							Checkout
						</button>
					</div>
					<ErrorMessage name="amount" render={renderError} />
				</div>
			</Form>

		</Formik>

	)
}
