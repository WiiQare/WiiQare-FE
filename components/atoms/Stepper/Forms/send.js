import React, { useContext } from "react";
import { ErrorMessage, Field, Form, Formik } from "formik";
import { FormContext } from "../../../../pages/voucher/buy";
import * as yup from "yup";
import { HiOutlineInformationCircle } from "react-icons/hi";

function Send() {
	const { activeStepIndex, setActiveStepIndex, formData, setFormData } = useContext(FormContext);

	const renderError = (message) => (
		<p className="text-xs text-red-600 font-light flex items-center gap-1"><HiOutlineInformationCircle />{message}</p>
	);

	const ValidationSchema = yup.object().shape({
		
	});

	return (
		<Formik
			initialValues={{
				
			}}
			validationSchema={ValidationSchema}
			onSubmit={(values) => {
				const data = { ...formData, ...values };
				setFormData(data);
				setActiveStepIndex(activeStepIndex + 1);
			}}
		>
			<Form className="flex flex-col justify-start md:w-2/3 gap-4">
				<div className="text-2xl font-medium my-4 capitalize">Send Voucher !</div>
				
                <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Fuga, vero nulla doloribus harum, ipsa reprehenderit delectus minus excepturi inventore, iste in suscipit? Neque facere laudantium dicta optio ullam omnis totam.
                </p>

				<div className="flex flex-row-reverse ">
                    <button
                        className="bg-primary flex gap-3 items-center w-fit font-medium text-white my-2 py-3 px-5 hover:bg-blue-500 duration-200 transition-all hover:shadow rounded-lg"
                        type="submit"
                    >
                        Send Voucher
                    </button>
				</div>
			</Form>
		</Formik>
	);
}

export default Send;
