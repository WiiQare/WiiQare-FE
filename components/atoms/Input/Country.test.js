import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Country from "./Country";
import { CountryContext } from "../Stepper/Forms/identity2";

describe("Country", () => {
  let component;
  beforeEach(() => {
    const res = render(
      <CountryContext.Provider value={{ country: "Indonesia" }}>
        <Country />
      </CountryContext.Provider>
    );
    component = res.container;
  });

  test("should render Country component", () => {
    expect(component).toMatchSnapshot();
  });
});
