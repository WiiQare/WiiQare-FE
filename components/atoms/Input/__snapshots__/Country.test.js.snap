// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country should render Country component 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon w-full css-l3ln04-MuiAutocomplete-root"
    contenteditable="false"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="country-select-demo"
        id="country-select-demo-label"
      >
        Choisissez le pays
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="new-password"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id="country-select-demo"
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Choisissez le pays
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
